packages/postgres-database/tests/test_utils_folders_v2.py

```diff
@@ -0,0 +1,1856 @@
+from simcore_postgres_database.utils_folders_v2 import (
+    _FOLDER_NAME_MAX_LENGTH,
+    _FOLDER_NAMES_RESERVED_WINDOWS,
+    _ROLE_TO_PERMISSIONS,
+    EDITOR_PERMISSIONS,
+    NO_ACCESS_PERMISSIONS,
+    OWNER_PERMISSIONS,
+    VIEWER_PERMISSIONS,
+    CannotMoveFolderSharedViaNonPrimaryGroupError,
+    FolderAccessRole,
+    FolderAlreadyExistsError,
+    FolderEntry,
+    FolderNotFoundError,
+    FolderNotSharedWithGidError,
+    GroupIdDoesNotExistError,
+    InsufficientPermissionsError,
+    InvalidFolderNameError,
+    _FolderID,
+    _FolderPermissions,
+    _get_permissions_from_role,
+    _get_top_most_access_rights_entry,
+    _get_true_permissions,
+    _GroupID,
+    _ProjectID,
+    _requires,
+    folder_add_project,
+    folder_create,
+    folder_delete,
+    folder_list,
+    folder_move,
+    folder_remove_project,
+    folder_share_or_update_permissions,
+    folder_update,
+)
+from sqlalchemy.sql.elements import ColumnElement
+
+
+def test_permissions_integrity():
+    assert set(FolderAccessRole) == set(_ROLE_TO_PERMISSIONS.keys())
+
+
+@pytest.mark.parametrize(
+    "role, expected_permissions",
+    [
+        (FolderAccessRole.VIEWER, {"read": True, "write": False, "delete": False}),
+        (FolderAccessRole.EDITOR, {"read": True, "write": True, "delete": False}),
+        (FolderAccessRole.OWNER, {"read": True, "write": True, "delete": True}),
+    ],
+)
+def test_role_permissions(
+    role: FolderAccessRole, expected_permissions: dict[str, bool]
+):
+    assert _get_permissions_from_role(role) == expected_permissions
+
+
+@pytest.mark.parametrize(
+    "permissions, expected",
+    [
+        ([], {"read": False, "write": False, "delete": False}),
+        ([VIEWER_PERMISSIONS], {"read": True, "write": False, "delete": False}),
+        ([EDITOR_PERMISSIONS], {"read": True, "write": True, "delete": False}),
+        (
+            [EDITOR_PERMISSIONS, VIEWER_PERMISSIONS],
+            {"read": True, "write": True, "delete": False},
+        ),
+        ([OWNER_PERMISSIONS], {"read": True, "write": True, "delete": True}),
+        (
+            [OWNER_PERMISSIONS, EDITOR_PERMISSIONS],
+            {"read": True, "write": True, "delete": True},
+        ),
+        (
+            [OWNER_PERMISSIONS, EDITOR_PERMISSIONS, VIEWER_PERMISSIONS],
+            {"read": True, "write": True, "delete": True},
+        ),
+    ],
+)
+def test__requires_permissions(
+    permissions: list[_FolderPermissions] | None, expected: dict[str, bool]
+):
+    assert _requires(*permissions) == expected
+
+
+@pytest.mark.parametrize(
+    "invalid_name",
+    [
+        None,
+        "",
+        "/",
+        ":",
+        '"',
+        "<",
+        ">",
+        "\\",
+        "|",
+        "?",
+        "My/Folder",
+        "MyFolder<",
+        "CON",
+        "AUX",
+        "My*Folder",
+        "A" * (_FOLDER_NAME_MAX_LENGTH + 1),
+        *_FOLDER_NAMES_RESERVED_WINDOWS,
+    ],
+)
```

Comment:
how are these restrictions set up? why do we have them?

Reply:
Future failsafe. If we ever want to mount these as "folders" on a disk. This way we have no issues with names.